hack/update-godep-licenses.sh

```diff
@@ -16,30 +15,187 @@
 
 # Update the Godeps/LICENSES.md document.
 # Generates a table of Godep dependencies and their license.
-# Requires:
-#    docker
-#    mesosphere/godep-licenses (docker image) - source: https://github.com/mesosphere/godep-licenses
+#
 # Usage:
-#    Run every time a license file is added/modified within /Godeps to update /Godeps/LICENSES.md.
-#    Add exceptions (-e <repo>:<license>) for any dependency (project) vendored by Godep
-#      that has a known license that isn't vendored by Godep or can't be found by godep-licenses.
+#    $0 [--create-missing] [/path/to/licenses]
+#
+#    --create-missing will write the files that only exist upstream, locally.
+#    This option is mostly used for testing as we cannot check-in any of the
+#    additionally created files into the godep auto-generated tree.
+#
+#    Run every time a license file is added/modified within /Godeps to
+#    update /Godeps/LICENSES
 
 set -o errexit
 set -o nounset
 set -o pipefail
 
-KUBE_ROOT="${KUBE_ROOT:-$(cd "$(dirname "${BASH_SOURCE}")/.." && pwd -P)}"
+###############################################################################
+# Manage the state of LICENSE/COPYRIGHT files
+# Default operation is to check to see if a file is in the state file.
+#
+# @optparam -a    Add the file to the state file
+# @param    file  The file to check or add
+# @return 1 when no file is found in state file
+#
+file_state () {
+  local add=0
+  case "$1" in
+    -a) add=1;shift ;;
+  esac
+  local file=$1
+
+  # If we're ignoring state, then return 1
+  ((CREATE_MISSING)) && return 1
+
+  # initialize if step 1
+  if ((add)); then
+    echo "$file" >> $GODEPS_STATE
+    return 0
+  fi
+
+  # Get return code from grep itself
+  # Redirect stderr so that a missing state file returns 1 quietly
+  LANG=C egrep -wq "^$file$" $GODEPS_STATE 2>/dev/null
+}
+
+###############################################################################
+# Process package content
+#
+# @param package  The incoming package name
+# @param type     The type of content (LICENSE or COPYRIGHT)
+#
+process_content () {
+  local package=$1 type=$2 f remote_files package_root ensure_pattern \
+        package_root_url dir_root find_maxdepth find_names
+  local -a local_files=()
+
+  # Necessary to expand {}
+  case $type in
+      LICENSE) remote_files=$(echo LICENSE{,.code,.txt,.md})
```

Comment:
Prefer arrays when possible. This is also a useless echo. So:
```
remote_files=( NOTICE{,.txt} README{,.md} )
```
(and fix up uses)